### DISCUSSION TITLE
transcribe.py decode_with_fallback() only expects a single DecodingResult from model.decode() but it can be a list of DecodingResult

### DISCUSSION THREAD

**Discussion by RealHandy** [openai/whisper#2333] [Q&A]
Trying to transcribe a tensor value, I get
```
File "whisper\transcribe.py", line 201, in decode_with_fallback
    and decode_result.compression_ratio > compression_ratio_threshold
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'list' object has no attribute 'compression_ratio'
```
which I've determined is because decode_result can be a DecodingResult or a List[DecodingResult], but the erroring line in transcribe.py, and in fact decode_with_fallback() in its entirety, doesn't expect model.decode() to return a list. Is this my problem for passing a tensor that can generate a list in model.decode(), or is this a bug?

Thanks,
Andy

**Comment by noahvandal** [comment]
I had a similar issue. I was able to fix the problem by adding these lines after the `decode_result` output: 

`if isinstance(decode_result, list):
    decode_result = decode_result[0]`

at line 196 of whisper/transcribe.py. A PR could be made for this, unless we are using this wrong.

**Comment by ftyers** [comment]
This also worked for me, following the code on [this page](https://medium.com/@david.richards.tech/how-to-build-a-streaming-whisper-websocket-service-1528b96b1235), but it seems like we shouldn't need to make a change to the library code for this use case...
